File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maxSubarrayLength = function(nums, k) {
    const freq = {}
    let l = 0, res = 0
    for (let r=0; r < nums.length; r++) {
        freq[nums[r]] = freq[nums[r]]+ 1 || 1
        while (l < r && freq[nums[r]] > k) {
            freq[nums[l]]--
            l++
        }
        res = Math.max(res, r - l + 1)
    }
    return res
};

File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        freq = {}
        l = res = 0
        for r in range(0, len(nums)):
            freq[nums[r]] = freq.get(nums[r], 0) + 1
            while l < r and freq[nums[r]] > k:
                freq[nums[l]] -= 1
                l += 1
            res = max(res, r - l + 1)
        return res